activemodel/lib/active_model/errors.rb

```diff
@@ -258,13 +303,30 @@ def as_json(options = nil)
     #   person.errors.to_hash       # => {:name=>["cannot be nil"]}
     #   person.errors.to_hash(true) # => {:name=>["name cannot be nil"]}
     def to_hash(full_messages = false)
-      if full_messages
-        messages.each_with_object({}) do |(attribute, array), messages|
-          messages[attribute] = array.map { |message| full_message(attribute, message) }
-        end
-      else
-        without_default_proc(messages)
+      hash = {}
+      message_method = full_messages ? :full_message : :message
+      group_by_attribute.each do |attribute, errors|
+        hash[attribute] = errors.map(&message_method)
+      end
+      hash
+    end
+
+    def messages
+      DeprecationHandlingMessageHash.new(self)
+    end
+
+    def details
+      hash = {}
```

Comment:
I believe this hash needs to have an empty array as default. Otherwise `foo.errors.defails[:base].any?` will now fail.